pkg/resmgr/cache/cache.go

```diff
@@ -1027,6 +998,10 @@ func (cch *cache) mkdirAll(what, path string, p *permissions) error {
 	return nil
 }
 
+func (cch *cache) SerializeAsJSON() ([]byte, error) {
```

Comment:
nitpick: should/could we use `cache.Snapshot()` here directly ? It is already exported. Or rename it if we don't like the name anymore ?

Also, if we've now nuked all code paths that might externally lock the cache and then invoke (indirectly) `cache.Snapshot()` (via `cache.Save()`), then I think we should lock the cache now while we take the snapshot. The command line option is always present. When enabled this function can be triggered externally. And while taking the snapshot, we iterate over the container and pod maps.